### PR TITLE
Fix story stretched poster images by removing media pool default sources.

### PR DESCRIPTION
**Pull request by gmajoulet**
1. Why the poster images are stretched
If a video poster image is displayed and you call `play()` before having enough data to even play the first frame, you get a stretched poster image. It's a browser thing.

2. Why we call `play()` too early
We're not supposed to, we invested a lot on the `loadPromise()` which should protect us from that.
However the `loadPromise()` has an early return when it detects that the element is already loaded. Sometimes (but not always), the default sources lead to the media element's `readyState` to be `4`, which resolves our `loadPromise()` and calls `play()` too early, stretching the poster.

3. Solution
There are several things we could do to fix this issue, like ensuring we set the new sources and called `load()` on the `HTMLMediaElement` before calling the `loadPromise()`. This would be some refactoring and idk how straightforward it would be as the media pool is making certain assumptions in `swapInDom`.
Instead, we're going with an easier fix and removing the default sources, which saves some bytes from the story extension.

Fixes https://github.com/ampproject/amphtml/issues/30651

### CODE REVIEW

File: extensions/amp-story/1.0/media-pool.js [hunk 1]
```diff
@@ -546,13 +516,14 @@ export class MediaPool {
...
     ).then(
       () => {
-        this.maybeResetAmpMedia_(ampMediaForPoolEl);
-        this.maybeResetAmpMedia_(ampMediaForDomEl);
         this.enqueueMediaElementTask_(
           poolMediaEl,
           new UpdateSourcesTask(this.win_, sources)
         );
-        this.enqueueMediaElementTask_(poolMediaEl, new LoadTask());
+        this.enqueueMediaElementTask_(poolMediaEl, new LoadTask()).then(() => {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/media-pool.js:523]: An additional fix here, `maybeResetAmpMedia_` will call `resetOnDomChange` that will call `loadPromise()`. So we don't make this call until the `loadTask` ran, since it will restart the media selection process and `readyState`.

**Comment by newmuis** [comment]
Can you run through the video manual QA test suite on Android Chrome, iOS Safari, and Android Firefox to confirm they pass?  (I can help if you need browser/device coverage)

File: extensions/amp-story/1.0/media-pool.js (hunk 1 above)
Comment (newmuis) [reply extensions/amp-story/1.0/media-pool.js:523]: Just to be super sure since this being false could introduce really hairy race conditions: Is the expected behavior that calling `loadPromise` either before or after the video has loaded should eventually resolve?  i.e. we don't have to worry about the case where the video has already loaded before this `then` block gets executed?

Comment (gmajoulet) [reply extensions/amp-story/1.0/media-pool.js:523]: Yep, the `loadPromise` is pretty smart about this!

This was actually the issue, `loadPromise` was detecting that the default sources loaded, not the actual ones, and calling play() too early. Not having default sources fixes the issue, and I added an extra fix to call load() when removing the sources so it resets the `readyState` and all we care about.

**Comment by gmajoulet** [comment]
Can you double check on your Android device? https://stamp-stretched-posters.firebaseapp.com/examples/qa/video.html
I think it's looking good!

PTAL

**Review by Enriqe — COMMENTED** [review]
Why did we have the default sources in the first place? don't we get errors/warnings if there's no src on the element?

File: extensions/amp-story/1.0/media-pool.js [hunk 2]
```diff
@@ -588,13 +559,12 @@ export class MediaPool {
...
   resetPoolMediaElementSource_(poolMediaEl) {
-    const mediaType = this.getMediaType_(poolMediaEl);
-    const defaultSources = this.getDefaultSource_(mediaType);
+    const defaultSources = this.getDefaultSource_();
 
     return this.enqueueMediaElementTask_(
       poolMediaEl,
       new UpdateSourcesTask(this.win_, defaultSources)
-    );
+    ).then(() => this.enqueueMediaElementTask_(poolMediaEl, new LoadTask()));
```
Comment (Enriqe) [inline extensions/amp-story/1.0/media-pool.js:567]: why is this needed?

File: extensions/amp-story/1.0/media-pool.js [hunk 3]
```diff
@@ -194,19 +192,6 @@ export class MediaPool {
      */
     this.blessed_ = false;
 
-    /**
-     * The default source to use for pool-created audio sources,
-     * @private @const {!Object<!MediaType, string>}
-     */
-    this.defaultSources_ = {
-      [MediaType.AUDIO]: isExperimentOn(win, 'disable-amp-story-default-media')
```
Comment (Enriqe) [inline extensions/amp-story/1.0/media-pool.js:202]: do we have to delete the experiment from the config or something?

File: extensions/amp-story/1.0/media-pool.js (hunk 2 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/media-pool.js:567]: Gave more context offline, but for the record:

>  I added an extra fix to call load() when removing the sources so it resets the readyState and all we care about.

File: extensions/amp-story/1.0/media-pool.js (hunk 3 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/media-pool.js:202]: I believe experiments usage and declarations have to be removed in separate PRs

**Review by newmuis — APPROVED** [review]
QA passed on my Pixel 2.

As a side note: We should probably also add a story with `N` autoplay videos where `N > size of mediapool for videos`.  A lot of failure cases are around autoplaying media and around mediapool recycling, and this would force both.

File: extensions/amp-story/1.0/media-pool.js (hunk 3 above)
Comment (kristoferbaxter) [reply extensions/amp-story/1.0/media-pool.js:202]: Let's remove the experiment quickly though before it's forgotten about.

File: extensions/amp-story/1.0/media-pool.js [hunk 4]
```diff
@@ -546,13 +516,14 @@ export class MediaPool {
...
-        this.maybeResetAmpMedia_(ampMediaForDomEl);
         this.enqueueMediaElementTask_(
           poolMediaEl,
           new UpdateSourcesTask(this.win_, sources)
         );
-        this.enqueueMediaElementTask_(poolMediaEl, new LoadTask());
+        this.enqueueMediaElementTask_(poolMediaEl, new LoadTask()).then(() => {
+          this.maybeResetAmpMedia_(ampMediaForPoolEl);
+          this.maybeResetAmpMedia_(ampMediaForDomEl);
+        });
```
Comment (kristoferbaxter) [inline extensions/amp-story/1.0/media-pool.js:526]: This is another example where `await` will make code easier to read.

Looking forward to seeing the refactor afterwards.

File: extensions/amp-story/1.0/media-pool.js (hunk 2 above)
Comment (kristoferbaxter) [reply extensions/amp-story/1.0/media-pool.js:567]: Could you leave a comment to this? This set of functionality could use with some additional documentation.

**Review by kristoferbaxter — COMMENTED** [review]
Approved. But some additional test coverage would be quite valuable.

File: extensions/amp-story/1.0/media-pool.js (hunk 3 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/media-pool.js:202]: https://github.com/ampproject/amphtml/pull/31111
